### PR TITLE
Use `trackWithSiteDetails` to track failed media uploads in the editor 

### PR DESCRIPTION
**Pull request by jhnstn**
This a follow up to #19853 

The changes here track `editor_upload_media_*` events.

-----

## To Test:

- Turn on Airplane mode on the device
- Open the editor and add an image from device to a post
- Notice that the `editor_upload_media_failed` event in the app log includes the site info: `site_type`, `blog_id`, and `is_jetpack`
- Hit retry on the block and confirm the retry modal 
- - Notice that the `editor_upload_media_retried` event in the app log includes the site info: `site_type`, `blog_id`, and `is_jetpack`
-----

## Regression Notes

1. Potential unintended areas of impact

    Change in metric tracking 

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Verified that only new keys are added to the tracking event
    - Verified that the event ids are unchanged 

3. What automated tests I added (or what prevented me from doing so)

    No additional test coverage is needed.

-----

## PR Submission Checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

-----
## UI Changes Testing Checklist:

- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by SiobhyB — APPROVED** [review]
Thanks @jhnstn, I confirmed that, following the changes from this PR, `blogid`, `site_type`, and `is_jetpack` are now sent with the `jpandroid_editor_upload_media_failed` and `jpandroid_editor_upload_media_retried` events. 

I also confirmed that `blog_id` and `is_jetpack` are included when the following events are fired (`site_type` was already there for these events):

* `jpandroid_media_service_upload_started`
* `jpandroid_media_service_upload_response_error`
* `jpandroid_media_service_upload_response_ok`

As per our discussion in the comments in https://github.com/wordpress-mobile/WordPress-Android/pull/19853, I'm not sure why these events weren't previously showing the expected props for me, but they are now. 🎉

File: WordPress/src/main/java/org/wordpress/android/ui/posts/editor/media/RetryFailedMediaUploadUseCase.kt [hunk 1]
```diff
@@ -33,7 +39,8 @@ class RetryFailedMediaUploadUseCase @Inject constructor(
                         editorMediaListener,
                         mediaModels
                     )
-                    tracker.track(Stat.EDITOR_UPLOAD_MEDIA_RETRIED)
+                    val site: SiteModel? = siteStore.getSiteByLocalId(editorMediaListener.getImmutablePost().localSiteId)
```
Comment (SiobhyB) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/editor/media/RetryFailedMediaUploadUseCase.kt]: It looks like this line just needs to be reformatted, the linter's failing with the following:

```
[2024-01-08T23:57:45Z] /var/lib/buildkite-agent/builds/ci-android-i-0a5985984fd2646ad-1/automattic/wordpress-android/WordPress/src/main/java/org/wordpress/android/ui/posts/editor/media/RetryFailedMediaUploadUseCase.kt:42:1: Line detected, which is longer than the defined maximum line length in the code style. [MaxLineLength]
```

File: WordPress/src/main/java/org/wordpress/android/ui/posts/editor/media/RetryFailedMediaUploadUseCase.kt [hunk 2]
```diff
@@ -3,7 +3,11 @@ package org.wordpress.android.ui.posts.editor.media
 import dagger.Reusable
 import org.wordpress.android.analytics.AnalyticsTracker.Stat
 import org.wordpress.android.fluxc.model.MediaModel.MediaUploadState.QUEUED
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.fluxc.store.SiteStore
+import org.wordpress.android.ui.CommentFullScreenDialogFragment
```
Comment (SiobhyB) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/editor/media/RetryFailedMediaUploadUseCase.kt]:
```suggestion
```

I don't think this is needed, as per the failing test:

```
[2024-01-08T23:57:45Z] /var/lib/buildkite-agent/builds/ci-android-i-0a5985984fd2646ad-1/automattic/wordpress-android/WordPress/src/main/java/org/wordpress/android/ui/posts/editor/media/RetryFailedMediaUploadUseCase.kt:8:1: The import 'org.wordpress.android.ui.CommentFullScreenDialogFragment' is unused. [UnusedImports]
```

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by SiobhyB — APPROVED** [review]
@jhnstn, this looks good to me! I see the `buildkite/wordpress-android/unit-test-wordpress` test is failing with an unclear error, I wonder if updating from `trunk` may resolve it? 🤔 Regardless, I confirmed the events work as expected for me, thank you!

**Comment by AjeshRPai** [comment]
Hey @jhnstn 👋🏼 
Since I am a little occupied with project work + @SiobhyB has reviewed the PR. I am not reviewing this PR. Feel free to let me know if you also want a review from me. Otherwise, Please feel free to merge at your convenience.

**Comment by SiobhyB** [comment]
@jhnstn, it looks like the test failures are legitimate, I'm seeing the following errors if I further expand the results:

```
[2024-01-15T20:37:30Z] org.wordpress.android.ui.posts.editor.media.RetryFailedMediaUploadUseCaseTest > Sets upload state to QUEUED to all media models FAILED
--
  | [2024-01-15T20:37:30Z]     java.lang.NullPointerException at RetryFailedMediaUploadUseCaseTest.kt:39
  | [2024-01-15T20:37:30Z]
  | [2024-01-15T20:37:30Z] org.wordpress.android.ui.posts.editor.media.RetryFailedMediaUploadUseCaseTest > Tracks upload retried event FAILED
  | [2024-01-15T20:37:30Z]     java.lang.NullPointerException at RetryFailedMediaUploadUseCaseTest.kt:72
  | [2024-01-15T20:37:30Z]
  | [2024-01-15T20:37:30Z] org.wordpress.android.ui.posts.editor.media.RetryFailedMediaUploadUseCaseTest > Invokes save and initiate upload on all posts FAILED
  | [2024-01-15T20:37:30Z]     java.lang.NullPointerException at RetryFailedMediaUploadUseCaseTest.kt:56
  | [2024-01-15T20:37:30Z]
  | [2024-01-15T20:37:30Z] org.wordpress.android.ui.posts.editor.media.RetryFailedMediaUploadUseCaseTest > Loads all failed media from db FAILED
  | [2024-01-15T20:37:30Z]     java.lang.NullPointerException at RetryFailedMediaUploadUseCaseTest.kt:27
```

**Comment by jhnstn** [comment]
@SiobhyB thanks for checking out those errors. 
I had to do a little bit of refactoring and update the tests to get it working. Changes are in https://github.com/wordpress-mobile/WordPress-Android/pull/19900/commits/825e67e4e2053d7ea03f28e4f086e0676bea7fbb

The tests were only failing because my changes had to be mocked in the tests.

I had auto merge enabled but I was going to have you check out the refactor change. It's not much of change so I think it's fine.

**Comment by SiobhyB** [comment]
@jhnstn, perfect, the changes look good to me! Thank you for merging!

**Pull request merged**